src/services_models/services_constructor.jl

```diff
@@ -7,8 +7,8 @@ function construct_services!(
     isempty(services_template) && return
     for service_model in values(services_template)
         @debug "Building $(service_model.service_type) with $(service_model.formulation) formulation"
-        services = PSY.get_components(service_model.service_type, sys)
-        if validate_available_services(service_model.service_type, services)
+        services = service_model.service_type[]
+        if validate_available_services(service_model.service_type, services, sys)
```

Comment:
Consider renaming this to `validate_available_services!` so that it's more obvious that you're mutating `services`.